gensim/matutils.py

```diff
@@ -353,13 +371,14 @@ def unitvec(vec):
     except:
         return vec
 
-    if isinstance(first, (tuple, list)) and len(first) == 2: # gensim sparse format?
-        length = 1.0 * math.sqrt(sum(val ** 2 for _, val in vec))
-        assert length > 0.0, "sparse documents must not contain any explicit zero entries"
-        if length != 1.0:
-            return [(termid, val / length) for termid, val in vec]
-        else:
-            return list(vec)
+    if isinstance(first, (tuple, list)) and len(first) == 2: # gensim sparse format
+        if norm == 'l1':
+            length = float(sum(abs(val) for _, val in vec))
+            return ret_normalized_vec(vec, length)
+        if norm == 'l2':
+            length = 1.0 * math.sqrt(sum(val ** 2 for _, val in vec))
+            assert length > 0.0, "sparse documents must not contain any explicit zero entries"
```

Comment:
Should I change this to returning the vector unchanged in case of zero length as we are doing with other input types?